integration_tests/e2e/addPrisonerNonAssociation.cy.ts

```diff
@@ -0,0 +1,53 @@
+import { davidJones, andrewBrown } from '../../server/data/testData/offenderSearch'
+import Page from '../pages/page'
+import PrisonerNonAssociations from '../pages/nonAssociations/prisonerNonAssociations'
+import addPrisonerNonAssociations from '../pages/nonAssociations/addPrisonerNonAssociations'
+import addPrisonerNonAssociationDetails from '../pages/nonAssociations/addPrisonerNonAssociationDetails'
+import addPrisonerNonAssociationConfirmation from '../pages/nonAssociations/addPrisonerNonAssociationConfirmation'
+
+context('Add prisoner non associations page', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+    cy.task('stubNomisUserCaseloads')
+    cy.task('stubOffenderSearchGetPrisonerResult', { prisonerNumber: 'A1234BC', result: davidJones })
+    cy.task('stubListNonAssociations')
+    cy.signIn()
+    cy.visit('/prisoner/A1234BC/non-associations')
+  })
+
+  it('navigate to add non association', () => {
+    const homePage = Page.verifyOnPage(PrisonerNonAssociations)
+    homePage.getAddNewNonAssociation().click()
+    cy.title().should('eq', `Search for a prisoner`)
+  })
+
+  it('should allow adding a new non association', () => {
+    cy.task('stubOffenderSearchResults', {
+      prisonId: 'MDI',
+      term: 'Andrew',
+      results: [andrewBrown],
+    })
+
+    const homePage = Page.verifyOnPage(PrisonerNonAssociations)
+    homePage.getAddNewNonAssociation().click()
+    cy.task('stubOffenderSearchGetPrisonerResult', { prisonerNumber: 'A5678CS', result: andrewBrown })
+
+    const addPage = Page.verifyOnPage(addPrisonerNonAssociations)
+    addPage.getInputField('search').type('Andrew')
```

Comment:
the `getInputField` method doesn't accept any parameters so `'search'` is actually entirely ignored